src/components/APIKeyModal/APIKeyModal.test.js

```diff
@@ -0,0 +1,40 @@
+import React from "react";
+import { fireEvent, render, screen } from "@testing-library/react";
+import { APIKeyModal } from "./APIKeyModal";
+import { axe, toHaveNoViolations } from "jest-axe";
+window.open = jest.fn(); // mock window.open for testing the buttons
+
+describe("Test the API key request modal", () => {
+  test("renders the API key request modal to test...", () => {
+    window.open.mockClear();
+    const query = render(<APIKeyModal />);
+
+    const { container, getByText } = query;
+    const apikey = screen.getByText("Request an API Key");
+    expect(apikey).toBeTruthy();
+    fireEvent.click(getByText("Request an API Key"));
+    const genAPI = screen.getByText(/Generate an API/i);
+    expect(genAPI).toBeInTheDocument();
+    const signUp = screen.getByText(/Sign up for an application/i);
```

Comment:
I don't think you need to test that every field renders, but that's fine. Try adding this `jest.setTimeout(30000);` to the top of the file like Mike has in the About.test.js test.